From c00b7ce8ba08abb618178090377c6a29e50e259b Mon Sep 17 00:00:00 2001
From: Matti Kosola <matti.kosola@jolla.com>
Date: Tue, 3 Sep 2019 15:10:03 +0000
Subject: [PATCH] sailfishos: Add sepolicy class for dbus

Running SELinux in Sailfish OS needs sepolicy class for D-Bus. Without
dbus class D-Bus cannot start when SELinux is enabled.

Change-Id: I50803a20ac3ee4c6c52de7db9615fa376a944327
Signed-off-by: Matti Kosola <matti.kosola@jolla.com>
---
 prebuilts/api/28.0/private/access_vectors   | 5 +++++
 prebuilts/api/28.0/private/security_classes | 2 ++
 private/access_vectors                      | 5 +++++
 private/security_classes                    | 2 ++
 4 files changed, 14 insertions(+)

diff --git a/prebuilts/api/28.0/private/access_vectors b/prebuilts/api/28.0/private/access_vectors
index 898c884c..ba25d3e5 100644
--- a/prebuilts/api/28.0/private/access_vectors
+++ b/prebuilts/api/28.0/private/access_vectors
@@ -724,3 +724,8 @@ class drmservice {
 	finalizeDecryptUnit
 	pread
 }
+
+class dbus {
+	acquire_svc
+	send_msg
+}
diff --git a/prebuilts/api/28.0/private/security_classes b/prebuilts/api/28.0/private/security_classes
index 251b7216..4b519e64 100644
--- a/prebuilts/api/28.0/private/security_classes
+++ b/prebuilts/api/28.0/private/security_classes
@@ -144,3 +144,5 @@ class keystore_key              # userspace
 
 class drmservice                # userspace
 # FLASK
+
+class dbus
diff --git a/private/access_vectors b/private/access_vectors
index 898c884c..ba25d3e5 100644
--- a/private/access_vectors
+++ b/private/access_vectors
@@ -724,3 +724,8 @@ class drmservice {
 	finalizeDecryptUnit
 	pread
 }
+
+class dbus {
+	acquire_svc
+	send_msg
+}
diff --git a/private/security_classes b/private/security_classes
index 251b7216..4b519e64 100644
--- a/private/security_classes
+++ b/private/security_classes
@@ -144,3 +144,5 @@ class keystore_key              # userspace
 
 class drmservice                # userspace
 # FLASK
+
+class dbus
-- 
2.17.1

